py/argcheck.c

```diff
@@ -29,49 +29,55 @@
 
 #include "py/runtime.h"
 
-void mp_arg_check_num(size_t n_args, size_t n_kw, size_t n_args_min, size_t n_args_max, bool takes_kw) {
+int mp_arg_check_num(size_t n_args, size_t n_kw, size_t n_args_min, size_t n_args_max, bool takes_kw) {
     // TODO maybe take the function name as an argument so we can print nicer error messages
 
     if (n_kw && !takes_kw) {
         if (MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE) {
             mp_arg_error_terse_mismatch();
         } else {
-            mp_raise_TypeError("function does not take keyword arguments");
+            mp_raise_TypeError_o("function does not take keyword arguments");
         }
+        return 1;
```

Comment:
I was waiting until it was all converted to see what the best thing to return would be for functions like this.  It might be best to use errno-like codes, -1 for error, 0 for success, etc.  Then there can be common wrappers/macros to deal with it all.